Soundcloud doesn't load any songs

### Checklist

- [X] I am able to reproduce the bug with the [latest version](https://github.com/TeamNewPipe/NewPipe/releases/latest).
- [X] I made sure that there are *no existing issues* - [open](https://github.com/TeamNewPipe/NewPipe/issues) or [closed](https://github.com/TeamNewPipe/NewPipe/issues?q=is%3Aissue+is%3Aclosed) - which I could contribute my information to.
- [X] I have read the [FAQ](https://newpipe.net/FAQ/) and my problem isn't listed.
- [X] I have taken the time to fill in all the required details. I understand that the bug report will be dismissed otherwise.
- [X] This issue contains only one bug.
- [X] I have read and understood the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/dev/.github/CONTRIBUTING.md).

### Affected version

0.24.0

### Steps to reproduce the bug

1. Open any song
2. Try to reproduce it

### Expected behavior

The song plays

### Actual behavior

It doesn't play the song and blames it on the network.

### Screenshots/Screen recordings


https://user-images.githubusercontent.com/114167756/198367329-e0a39436-04db-47fb-8b83-429c212a1e6f.mp4



### Logs

## Exception
* __User Action:__ play stream
* __Request:__ Player error[type=ERROR_CODE_PARSING_CONTAINER_MALFORMED] occurred while playing https://soundcloud.com/raowoul/from-y-to-y-len
* __Content Country:__ ES
* __Content Language:__ es-ES
* __App Language:__ es_ES
* __Service:__ SoundCloud
* __Version:__ 0.24.0
* __OS:__ Linux Android 12 - 31
<details><summary><b>Crash log </b></summary><p>

```
com.google.android.exoplayer2.ExoPlaybackException: Source error
	at com.google.android.exoplayer2.ExoPlayerImplInternal.handleIoException(ExoPlayerImplInternal.java:632)
	at com.google.android.exoplayer2.ExoPlayerImplInternal.handleMessage(ExoPlayerImplInternal.java:602)
	at android.os.Handler.dispatchMessage(Handler.java:102)
	at android.os.Looper.loopOnce(Looper.java:226)
	at android.os.Looper.loop(Looper.java:313)
	at android.os.HandlerThread.run(HandlerThread.java:67)
Caused by: com.google.android.exoplayer2.ParserException: Cannot find sync byte. Most likely not a Transport Stream.
	at com.google.android.exoplayer2.extractor.ts.TsExtractor.findEndOfFirstTsPacketInBuffer(TsExtractor.java:460)
	at com.google.android.exoplayer2.extractor.ts.TsExtractor.read(TsExtractor.java:327)
	at com.google.android.exoplayer2.source.hls.BundledHlsMediaChunkExtractor.read(BundledHlsMediaChunkExtractor.java:67)
	at com.google.android.exoplayer2.source.hls.HlsMediaChunk.feedDataToExtractor(HlsMediaChunk.java:473)
	at com.google.android.exoplayer2.source.hls.HlsMediaChunk.loadMedia(HlsMediaChunk.java:437)
	at com.google.android.exoplayer2.source.hls.HlsMediaChunk.load(HlsMediaChunk.java:394)
	at com.google.android.exoplayer2.upstream.Loader$LoadTask.run(Loader.java:412)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
	at java.lang.Thread.run(Thread.java:920)

```
</details>
<hr>


### Affected Android/Custom ROM version

Android 12

### Affected device model

Samsung galaxy s10

### Additional information

_No response_

Have the same issue but on a Pixel 3a. Works on wifi but not on data.

## Exception
* __User Action:__ play stream
* __Request:__ Player error[type=ERROR_CODE_PARSING_CONTAINER_MALFORMED] occurred while playing https://soundcloud.com/ssaliva/close2me-ss-stretchd_ascent-edit
* __Content Country:__ AU
* __Content Language:__ en-AU
* __App Language:__ en_AU
* __Service:__ SoundCloud
* __Version:__ 0.24.0
* __OS:__ Linux Android 12 - 32
<details><summary><b>Crash log </b></summary><p>

```
com.google.android.exoplayer2.ExoPlaybackException: Source error
	at com.google.android.exoplayer2.ExoPlayerImplInternal.handleIoException(ExoPlayerImplInternal.java:632)
	at com.google.android.exoplayer2.ExoPlayerImplInternal.handleMessage(ExoPlayerImplInternal.java:602)
	at android.os.Handler.dispatchMessage(Handler.java:102)
	at android.os.Looper.loopOnce(Looper.java:201)
	at android.os.Looper.loop(Looper.java:288)
	at android.os.HandlerThread.run(HandlerThread.java:67)
Caused by: com.google.android.exoplayer2.ParserException: Cannot find sync byte. Most likely not a Transport Stream.
	at com.google.android.exoplayer2.extractor.ts.TsExtractor.findEndOfFirstTsPacketInBuffer(TsExtractor.java:460)
	at com.google.android.exoplayer2.extractor.ts.TsExtractor.read(TsExtractor.java:327)
	at com.google.android.exoplayer2.source.hls.BundledHlsMediaChunkExtractor.read(BundledHlsMediaChunkExtractor.java:67)
	at com.google.android.exoplayer2.source.hls.HlsMediaChunk.feedDataToExtractor(HlsMediaChunk.java:473)
	at com.google.android.exoplayer2.source.hls.HlsMediaChunk.loadMedia(HlsMediaChunk.java:437)
	at com.google.android.exoplayer2.source.hls.HlsMediaChunk.load(HlsMediaChunk.java:394)
	at com.google.android.exoplayer2.upstream.Loader$LoadTask.run(Loader.java:412)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1137)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:637)
	at java.lang.Thread.run(Thread.java:1012)

```
</details>
<hr>

Sorry if i shouldnt comment on closed issues or have wrong formatting.

--- a/app/src/main/java/org/schabi/newpipe/util/ListHelper.java
+++ b/app/src/main/java/org/schabi/newpipe/util/ListHelper.java
@@ -189,13 +189,16 @@ public final class ListHelper {
 
     /**
      * Return a {@link Stream} list which only contains streams which can be played by the player.
-     * <br>
-     * Some formats are not supported. For more info, see {@link #SUPPORTED_ITAG_IDS}.
-     * Torrent streams are also removed, because they cannot be retrieved.
+     *
+     * <p>
+     * Some formats are not supported, see {@link #SUPPORTED_ITAG_IDS} for more details.
+     * Torrent streams are also removed, because they cannot be retrieved, like OPUS streams using
+     * HLS as their delivery method, since they are not supported by ExoPlayer.
+     * </p>
      *
      * @param <S>        the item type's class that extends {@link Stream}
      * @param streamList the original stream list
-     * @param serviceId
+     * @param serviceId  the service ID from which the streams' list comes from
      * @return a stream list which only contains streams that can be played the player
      */
     @NonNull
@@ -204,6 +207,8 @@ public final class ListHelper {
         final int youtubeServiceId = YouTube.getServiceId();
         return getFilteredStreamList(streamList,
                 stream -> stream.getDeliveryMethod() != DeliveryMethod.TORRENT
+                        && (stream.getDeliveryMethod() != DeliveryMethod.HLS
+                        || stream.getFormat() != MediaFormat.OPUS)
                         && (serviceId != youtubeServiceId
                         || stream.getItagItem() == null
                         || SUPPORTED_ITAG_IDS.contains(stream.getItagItem().id)));
@@ -295,7 +300,9 @@ public final class ListHelper {
         final Comparator<AudioStream> cmp = getAudioFormatComparator(context);
 
         for (final AudioStream stream : audioStreams) {
-            if (stream.getDeliveryMethod() == DeliveryMethod.TORRENT) {
+            if (stream.getDeliveryMethod() == DeliveryMethod.TORRENT
+                    || (stream.getDeliveryMethod() == DeliveryMethod.HLS
+                    && stream.getFormat() == MediaFormat.OPUS)) {
                 continue;
             }
 

